File: 프로그래머스/2/42842. 카펫/카펫.js
function solution(brown, yellow) {
    const all = brown + yellow;
    
    for(let i = 3; i <= Math.floor(all / 2); i++) {
        if(all % i !== 0) continue;
        
        const [x, y] = [all / i, i];
        
        if((x - 2) * (y - 2) === yellow) return [x, y];
    }
}

File: 프로그래머스/2/42842. 카펫/카펫.py
def yaksu(n):
    result = []
    for i in range(2, int(n**0.5) + 1):
        if n % i == 0:
            result.append(i)
    return result

def solution(brown, yellow):
    multiplied = brown + yellow
    for n in yaksu(multiplied):
        a, b = n, multiplied // n
        if (a - 2) * (b - 2) == yellow:
            return [b, a]